### ISSUE TITLE
Replace the request package by something else (DEPRECATED)

### ISSUE THREAD

**Issue by mahmoud-triki** [loopbackio/strong-soap#535]
# Description/Steps to reproduce
[Here ](https://www.npmjs.com/package/request)you find the deprecation warning, it has been deprecated for more than 3 years.

you can do npm install and you can see the deprecation warning.

# Link to reproduction sandbox
https://www.npmjs.com/package/request

# Expected result
Not using a deprecated package
This can be replaced by axios.

# Additional information

**Comment by villelahdenvuo** [comment]
Related: https://github.com/advisories/GHSA-p8p7-x288-28g6

I think replacement should be `node-fetch` and for Node 18 it should be optional since fetch is built in Node 18+.

**Comment by mahmoud-triki** [comment]
That's a good callout

**Comment by villelahdenvuo** [comment]
@dhmlau Hey, as this is a `Moderate` severity security issue that shows up on `npm audit` I wanted to make sure you see it. Seems like changing the http client to use fetch wouldn't require that much work.

**Comment by dhmlau** [comment]
@villelahdenvuo, I agree to use node-fetch. Would you like to submit a PR?

**Comment by villelahdenvuo** [comment]
@dhmlau We could keep backwards compatibility util Node 12 if we use https://github.com/nodejs/undici

**Comment by s100** [comment]
Please note that `strong-soap` also makes use of `request` indirectly, via `httpntlm-maa`. I have raised [an issue against `httpntlm-maa`](https://github.com/maa105/node-http-ntlm/issues/4), but the package has not been maintained for over three years, and I do not expect it to be fixed. So, please could you also consider stopping using `httpntlm-maa` or migrating away from it to something else.

**Comment by Timpan4** [comment]
Any updates on this matter?

**Comment by sseide** [comment]
> Please note that `strong-soap` also makes use of `request` indirectly, via `httpntlm-maa`. I have raised [an issue against `httpntlm-maa`](https://github.com/maa105/node-http-ntlm/issues/4), but the package has not been maintained for over three years, and I do not expect it to be fixed. So, please could you also consider stopping using `httpntlm-maa` or migrating away from it to something else.

@s100 Are you sure NPM releases are the version from maa105 and not the original one from Sam Decrock? https://github.com/SamDecrock/node-http-ntlm Maybe you created the ticket at the wrong repo...
But this repo non the less has no releases for a long time too - BUT Readme was updated "recently" with a new donation.

**Comment by s100** [comment]
> @s100 Are you sure NPM releases are the version from maa105 and not the original one from Sam Decrock?

Yes, I'm sure. Here's `strong-soap`'s dependency on `httpntlm-maa`:

https://github.com/loopbackio/strong-soap/blob/4a5a125e26ec70f10f9084e95d634ae1e4401de4/package.json#L11C6-L11C18

[Here's](https://www.npmjs.com/package/httpntlm-maa) the npm page for the `httpntlm-maa` package. Here's the repo link on that page:

![image](https://github.com/loopbackio/strong-soap/assets/9932290/82374484-2f65-4928-a909-d8465c49482d)

And that link goes [here](https://github.com/maa105/node-http-ntlm).

**Comment by sseide** [comment]
> > @s100 Are you sure NPM releases are the version from maa105 and not the original one from Sam Decrock?
> 
> Yes, I'm sure. Here's `strong-soap`'s dependency on `httpntlm-maa`:
> 
> https://github.com/loopbackio/strong-soap/blob/4a5a125e26ec70f10f9084e95d634ae1e4401de4/package.json#L11C6-L11C18
> 
> [Here's](https://www.npmjs.com/package/httpntlm-maa) the npm page for the `httpntlm-maa` package. Here's the repo link on that page:
> 
> ![image](https://user-images.githubusercontent.com/9932290/248220794-82374484-2f65-4928-a909-d8465c49482d.png)
> 
> And that link goes [here](https://github.com/maa105/node-http-ntlm).

oh - thanks. Need more coffee probably. I missed the small difference in the package name and only seen it is a fork of the other one with very similar readme file for booth packages...

Maybe the original one from Sam Decrock might be a possible replacement as development started again this year and it does not depend on "request" library. But contrary to the library from mma105 it does not support promisses, only callback style.

**Comment by ktj** [comment]
Consider using undici instead of fetch.
Fetch api is very limited since it's the same api that browsers implement.

**Comment by s100** [comment]
* Following #701, as of `strong-soap@4.0.2`, there is no longer a direct dependency on `request`.
* Following #717, as of `strong-soap@4.0.4`, there is no longer a direct dependency on `httpntlm-maa` either - this means there is no longer an indirect on `request` either.

This issue is fixed and can now be closed.
